source/blog/_posts/2018-08-23-against-microservices.md

```diff
@@ -0,0 +1,136 @@
+
+Or how do you secure that the negaotiated API interfaces actually work? Integration tests of course! All neighboring services must be integrated with one another, through the network, which means lots of tests for lots of services with lots of communication. 
+
+How will developers set up the application locally? Docker containers of course! Each Service runs in it's own container, comes usually at least one storage container. Somewhere is a routing configuration.
+
+Scaling contrary to common belief is an issue, too. There may be many independent Services that need to be scaled appropriately to improve the overall Performance of the application. 
+
+While all these issues are of course solvable, they will block development resources. They may delay the product significantly. They are the real **costs of Microservices**.   
+
+## The PHP factor
+
+So we are actually developing PHP here. PHP is single threaded, shared nothing often stateless function execution at it's heart. So when we talk about scaling we actually do not mean the application itself but the limits in the infrastructure. MySQL too slow? Use ElasticSearch! Webserver responds too slow? Put a load balancer in front of two! Horizontal Scaling is actually not that hard in PHP.
+
+Oh and while we are at it. The *Deployment Monolith Antipattern* is solvable through package management too. Just look at the shopware plugin store. PHP is an interpreted language, source changes in production are actually not the hardest problem.   
+
+## Benefits in cherry picking
+
+Of course I do not want to discard the whole notion of Microservices. There is value, but there are also other options. From a technical standpoint I would argue that projects usually do not fail because the test suite was too sophisticated, the documentation too helpful, or the architecture was too clear and predictable.
+
+And there are real world benefits to all of them! Good old static code analysis will help you greatly with dependency management and finally this can all be part of the global CI process.
+
+And it even might be necessary to create different Applications for parts of your problem. Mayber the requirements for a single part of your application are so drastically different that this becomes necessary. Feel free to use concepts from Microservices. **Hybridization** of architecture is not necessarily a bad thing.  
+
+Problem solved? Well there are still the organizational difficulties...
+
+## Alternative: (Micro-)Kernel
+
+Microservices and monoliths are on oposing sides of the static dependency spectrum. While a monolith is interconnected and interdependent a Microservice is *(almost)* not. Well this spectrum does not only consist of black and white. My favorite alternative is the kernel style. A Kernel represents a core domain of an application. In our case eCommerce. This kernel then provides an extension mechanism for others to extend, alter and replace these core concepts. A plugin system if you want.
```

Comment:
"oposing" => "opposing"

"and interdependent a Microservice..."  => "and an interdependent Microservice..."

"Kernel" => "kernel"